### PR TITLE
Gherkin and automation for PAC bootstrap form

### PR DESCRIPTION
**Pull request by sanketpathak**
Epic: https://issues.redhat.com/browse/ODC-6460
Story: https://issues.redhat.com/browse/ODC-6507
Acceptance criteria:

-  As an admin, I can follow a wizard-like experience to configure OpenShift and my git provider for pipeline as code

**Checks required for approving Epic gherkin scripts PR:**

- [x] Add @epic-number to the scenarios or feature file [e.g: @odc-xxx]
- [x] Add @to-do for automation possible scenarios
- [x] Add @regression or @smoke based on the importance of the scenario
- [x] Update the test scenarios count in [Automation status confluence Report](https://docs.jboss.org/display/ODC/Automation+Status+Report)
- [x] Check for the linter issues by executing `yarn run gherkin-lint` on frontend folder [Skip epic number tags related linter issues]

**Fix**: https://issues.redhat.com/browse/ODC-6508

**Test Setup**:

Cluster should be running on local

Check the TAGS in frontend/packages/pipelines/integration-tests/cypress.json file be: ""@pipelines and @[odc-6460](https://issues.redhat.com//browse/odc-6460) and not (@manual or @to-do or @un-verified or @broken-test)"",

Command to execute:

Run a cluster locally and execute the command `yarn run dev` and then `npm run test-cypress-pipelines` simultaneously

Execute the pipelines-as-code.feature file

**Browser**
Chrome 99

**Test Execution Screenshot:**

### CODE REVIEW

**Comment by hemantsaini-7** [comment]
image
/lgtm

File: frontend/packages/pipelines-plugin/integration-tests/features/pipelines/pipelines-as-code.feature [hunk 1]
```diff
@@ -132,3 +132,23 @@ Feature: Perform Actions on repository
...
+        #This test case is manual as it navigates to github in between the process
+        Scenario: Create and configure the GitHub Application to work with Pipelines as code: P-11-TC13
+            Given user is at Pipelines tab in admin page
+             When user clicks on Setup GitHub App button
+              And user enters GitHub application name as "pac-app123"
+              And user clicks on Setup button
+              And user confirms access in github
+              And user clicks Create GitHub App button in Create GitHub App page
+             Then user will be redirected to GitHub App details
+              And user will see App Name as "pac-app123", App Link as "https://github.com/apps/pac-app123" and Secret as "pipelines-as-code-secret"
```
Comment (karthikjeeyar) [inline frontend/packages/pipelines-plugin/integration-tests/features/pipelines/pipelines-as-code.feature:154]:
```suggestion
              And user will see App Name as "pac-app123", App Link as "https://github.com/apps/pac-app123" and Secret as "pipelines-as-code-secret" in "openshift-pipelines" namespace.
```

File: frontend/packages/pipelines-plugin/integration-tests/support/step-definitions/pipelines/pipeline-as-code.ts [hunk 2]
```diff
@@ -271,3 +274,22 @@ Then('user should see commit message in tooltip', () => {
...
 });
+
+Given('user is at Pipelines tab in admin page', () => {
+  perspective.switchTo(switchPerspective.Administrator);
+  tasksPage.openPipelinePage();
+  tasksPage.togglePipelineSidebar();
+});
+
+When('user clicks on Setup GitHub App button', () => {
+  cy.get('[data-test="secondary-action"]')
```
Comment (karthikjeeyar) [inline frontend/packages/pipelines-plugin/integration-tests/support/step-definitions/pipelines/pipeline-as-code.ts:285]: nit:
```suggestion
  cy.byTestID("secondary-action")
```

File: frontend/packages/pipelines-plugin/src/components/pac/PacAppForm.tsx [hunk 3]
```diff
@@ -30,7 +30,12 @@ const PacAppForm: React.FC<FormikProps<FormikValues>> = ({
     }
   };
   return (
-    <FlexForm action={PAC_GH_APP_NEW} onSubmit={submitFrom} method="post">
+    <FlexForm
+      action={PAC_GH_APP_NEW}
+      onSubmit={submitFrom}
+      method="post"
+      data-test="form-set-github-app"
```
Comment (karthikjeeyar) [inline frontend/packages/pipelines-plugin/src/components/pac/PacAppForm.tsx:37]:
```suggestion
      data-test="form-setup-github-app"
```

**Review by karthikjeeyar — APPROVED** [review]

File: frontend/packages/pipelines-plugin/integration-tests/support/step-definitions/pipelines/pipeline-as-code.ts [hunk 4]
```diff
@@ -271,3 +274,22 @@ Then('user should see commit message in tooltip', () => {
...
+    .should('be.visible')
+    .click();
+});
+
+Then('user can see {string}, {string} and {string}', (el1, el2, el3: string) => {
+  cy.get('[data-test="form-set-github-app"]')
```
[the remark this answers is not in the document]
Comment (sanketpathak) [reply frontend/packages/pipelines-plugin/integration-tests/support/step-definitions/pipelines/pipeline-as-code.ts]: Updated

File: frontend/packages/pipelines-plugin/src/components/pac/PacAppForm.tsx (hunk 3 above)
Comment (sanketpathak) [reply frontend/packages/pipelines-plugin/src/components/pac/PacAppForm.tsx]: Updated

File: frontend/packages/pipelines-plugin/integration-tests/support/step-definitions/pipelines/pipeline-as-code.ts (hunk 2 above)
Comment (sanketpathak) [reply frontend/packages/pipelines-plugin/integration-tests/support/step-definitions/pipelines/pipeline-as-code.ts]: Updated

**Comment by sanketpathak** [comment]
Pr only affects E2E tests
/label qe-approved
/label docs-approved
/label px-approved

**Pull request merged**